Make calo names configurable in DDCEDViewer

sfranchel opened the pull request:
BEGINRELEASENOTES
-  Configurable naming of calorimeters for  `DDCEDViewer`

ENDRELEASENOTES


As discussed in issue:
https://github.com/iLCSoft/CEDViewer/issues/29

this PR makes the naming of calorimeters configurable. This is required for compatibility with some recent designs of FCCee detectors.

Tagging @tmadlener

sfranchel committed:
set calo names configurable

tmadlener commented on include/DDCEDViewer.h:
```diff
@@ -181,7 +186,7 @@ CalorimeterDrawParams getCalorimeterParameters(dd4hep::Detector& theDetector, st
 //It suffices to perform the calculations in the first quadrant due to the detector's symmetry.
 //The signs of the tracks' directions are ultimately determined by the momenta.
-double calculateTrackLength(std::string type, dd4hep::Detector& theDetector, double x, double y, double z, double px, double py, double pz);
+double calculateTrackLength(std::string barrelName, std::string endcapName, dd4hep::Detector& theDetector, double x, double y, double z, double px, double py, double pz, double rel_X0 = 0.5);
```
Is it in any way possible to easily add `rel_X0` to the `CalorimeterDrawParams` and determine it from the `barrelName` or the `endcapName`? It looks to be rather tightly connected to that and it would make it much clearer. Additionally, all the complexity of determining that value could be "hidden" in `getCalorimeterParameters`.

sfranchel replied:
I'm not sure how that's expected to be determined. From my understanding, curently it is set to  `0.5` for both hcal and ecal caeses:
https://github.com/iLCSoft/CEDViewer/blob/master/src/DDCEDViewer.cc#L1204-L1218

While it is 0 only when drawing jets?
https://github.com/iLCSoft/CEDViewer/blob/master/src/DDCEDViewer.cc#L1107-L1116

tmadlener replied:
I think you are right, it seems to be more related to what is drawn, rather than where it's drawn. Can we revive this comment here?

https://github.com/iLCSoft/CEDViewer/blob/0a703fd35cfc77abb75f96d5d27a0d22e804e279/src/DDCEDViewer.cc#L1018

Maybe add a comment that it is used to make Jets appear differently.

tmadlener reviewed (commented):
Thanks a lot. Looks pretty sensible to me. I have one question (see below).

Once #31 is merged, could you pick up those changes? (An `update branch` button should appear) Then we would also have up-to-date CI fr this PR.

sfranchel committed:
Merge branch 'master' into dev-calo-naming

sfranchel committed:
add comment on rel_X0

sfranchel committed:
fix typo in comment

tmadlener commented on src/DDCEDViewer.cc:
```diff
@@ -1201,22 +1221,11 @@ CalorimeterDrawParams getCalorimeterParameters(dd4hep::Detector& theDetector, st
-        endcap = getCalorimeterParameters(theDetector, "ECalEndcap");
-        rel_X0 = 0.;
-    }
+//The rel_X0 refers to  mean interaction length traversing the material perpendicularly; must be <= 1 ! For drawing jets is set to 0.
```
Can we put this comment (as is) into the header file, where some more comments are? 

I also just realized that those are duplicated here and in the header file. I would just remove them in the .cc file and only keep the ones in the header file.

sfranchel replied:
updated now!

sfranchel committed:
move comments to header

tmadlener reviewed (approved):
Thanks a lot for addressing all the comments. Unfortunately real tests are a bit scarce in this part of Key4hep. Just to confirm: This version works for your purposes, right?

sfranchel commented:
> Thanks a lot for addressing all the comments. Unfortunately real tests are a bit scarce in this part of Key4hep. Just to confirm: This version works for your purposes, right?

Thank you @tmadlener for the help with this PR! I've managed to test locally the changes, and they do work as intended. :partying_face:
